Runtime/Managers/SelectedObjectsManager.cs

```
@@ -81,6 +78,7 @@ public void ClearSelection()
 
         public GameObject RemoveObjectFromSelection(GameObject selectedObject)
         {
+            if (!_allowSelection) return null;
```

We just talked about this, but I think making these kinds of sudden changes in the behavior of methods is a bad idea. and personally, I don't like when things fail silently.

Also, if it's true this is messing with the delete objects feature and we didn't catch that in a test, it means we have a test coverage deficiency in that flow.